server/index.ts

```diff
@@ -1,9 +1,31 @@
 import express from 'express';
-import path from 'path';
+// import path from 'path';
+import { connectDatabase } from './database';
+import { getUserCollection } from './database';
+import dotenv from 'dotenv';
+dotenv.config();
 
 const port = process.env.PORT || 3001;
 const app = express();
 
+if (!process.env.MONGODB_URI) {
+  throw new Error('No MongoDB URI dotenv variable');
+}
+
+connectDatabase(process.env.MONGODB_URI).then(() =>
+  app.listen(port, () => {
+    console.log(`Example app listening at http://localhost:${port}`);
+  })
+);
```

Comment:
I would move this to the bottom of the page to make sure, that all endpoints are added before start listening